Client/Frontend/Browser/TopTabsViewController.swift

```diff
@@ -328,44 +311,192 @@ extension TopTabsViewController: TabSelectionDelegate {
     func didSelectTabAtIndex(index: Int) {
         let tab = tabsToDisplay[index]
         tabManager.selectTab(tab)
-        collectionView.reloadData()
-        collectionView.setNeedsDisplay()
-        delegate?.topTabsDidChangeTab()
-        scrollToCurrentTab()
     }
 }
 
-extension TopTabsViewController : WKNavigationDelegate {
-    func webView(webView: WKWebView, didFinishNavigation navigation: WKNavigation!) {
-        collectionView.reloadData()
+// Collection Diff (animations)
+extension TopTabsViewController {
+
+    struct tableUpdate {
```

Comment:
Nit: Data structures should be capitalized. Also, I think naming this `TopTabChangeSet` would be more meaningful since `tableUpdate` is an action and not a noun.